celery/datastructures.py

```diff
@@ -695,10 +695,10 @@ def purge(self, now=None):
         if hasattr(now, '__call__'):
             now = now()  # if we got this now as function, evaluate it
         if self.maxlen:
+            inserted_time, _ = self._heap[0]
```

Comment:
this will have an error if self._data is empty

Reply:
I'm missing an if condition....